net/http/src/THttpServer.cxx

```diff
@@ -924,18 +828,19 @@ void THttpServer::ProcessRequest(THttpCallArg *arg)
          // if accepted, reply with connection id, which must be used in the following communications
          arg->SetMethod("WS_CONNECT");
 
-         if (handler && handler->HandleWS(arg)) {
-            arg->SetMethod("WS_READY");
-
-            TLongPollEngine *handle = new TLongPollEngine("longpoll", arg->fPathName.Data());
+         THttpLongPollEngine *handle = new THttpLongPollEngine("longpoll", arg->fPathName.Data());
+         arg->SetWSId(handle->GetId());
 
-            arg->SetWSId(handle->GetId());
+         if (handler->HandleWS(arg)) {
+            arg->SetMethod("WS_READY");
             arg->SetWSHandle(handle);
 
             if (handler->HandleWS(arg)) {
                arg->SetContent(TString::Format("%u", arg->GetWSId()));
```

Comment:
Does it make sense as an interface to set the content without setting the content type? I.e. should this be `SetContent(type, content)`?

Reply:
Content is special field in HTTP request. 
It can be binary or text form - both are supported in THttpCallArg. 
And content type is just another HTTP header field. 
Which one can modify without changing content itself.

Reply:
Splitting them allows setting the content but not type, or vice versa. That's bad.

Reply:
But content type is not always required.
It is just good practice, which helps browser to provide some default view for your reply.
But it is just optional field